alarms.tf

```diff
@@ -19,7 +19,7 @@ resource "aws_cloudwatch_metric_alarm" "alarm_rds_DatabaseConnections_writer" {
 }
 
 resource "aws_cloudwatch_metric_alarm" "alarm_rds_DatabaseConnections_reader" {
-  count               = "${var.enabled && var.cw_alarms && var.replica_count > 0 ? 1 : 0}"
+  count               = "${var.enabled ? var.cw_alarms && var.replica_count > 0 ? 1 : 0 : 0}"
```

Comment:
Is this necessary? I think the old way wouldn't create a resource if `enabled = false` and I find it was easier to read before (regular ternary vs double/nested ternary).

Reply:
If either cw_alarms is true or replica_count is greater than 0 it will create the alarm even if enabled is false.

This way I don't believe it will

Reply:
Normally `a && b && c` would require all 3 to be true.

Reply:
Yeah I agree, the >0 check is slightly confusing.

If var.replica_count is greater than 0 does it evaluate to true? Then the >0 check isn't actually needed?

Checking the Terraform doc && is a boolean check so I will change it back

Reply:
> If var.replica_count is greater than 0 does it evaluate to true? Then the >0 check isn't actually needed?

I think Terraform breaks if you try to evaluate integers as booleans like that. Python works the way you describe. I would leave `> 0` in either way, as it makes the logic a little bit clearer. I think how you had it before was best:

```
${var.enabled && var.cw_alarms && var.replica_count > 0 ? 1 : 0}
```